### PR TITLE
Revert Dispose in IIS tests

### PR DESCRIPTION
**Pull request by HaoK**
This seems to expose more unstability in the IIS functionals

### CODE REVIEW

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs [hunk 1]
```diff
@@ -391,11 +391,8 @@ private void Stop()
                 // Stop all app pools
                 foreach (var appPool in serverManager.ApplicationPools)
                 {
-                    if (appPool.State != ObjectState.Stopped && appPool.State != ObjectState.Stopping)
-                    {
-                        var state = appPool.Stop();
-                        Logger.LogInformation($"Stopping pool, state: {state}");
-                    }
+                    var state = site.Stop();
```
Comment (dougbu) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs:394]: Something wrong w/ conflict resolution here I assume. Are parts of b2830adc665b and 9d121881aaf4 causing problems❔

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/GlobalVersionTests.cs [hunk 2]
```diff
@@ -255,12 +255,12 @@ private static void CopyShimToOutput(IISDeploymentParameters parameters)
 
     private static void CopyFiles(DirectoryInfo source, DirectoryInfo target)
     {
-        foreach (var directoryInfo in source.GetDirectories())
+        foreach (DirectoryInfo directoryInfo in source.GetDirectories())
```
Comment (dougbu) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/GlobalVersionTests.cs:258]: bit of nice cleanup we're losing. anything important @HaoK❔

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs [hunk 3]
```diff
@@ -397,7 +397,7 @@ private void Stop()
                         Logger.LogInformation($"Stopping pool, state: {state}");
                     }
                 }
-
+
```
Comment (dougbu) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs:400]: nit: suggest we don't need any changes to this file at all

**Review by dougbu — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/GlobalVersionTests.cs (hunk 2 above)
Comment (HaoK) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/GlobalVersionTests.cs:258]: Nah just minor stuff, hoping to revert this revert someday anyways

**Comment by davidfowl** [comment]
Interesting...

**Comment by HaoK** [comment]
Yeah the disposing PR seems to create a new flaky failure where the IIS tests to get in a weird state where the servermanager was unable to find the proper site to start/stop, so there must be some race condition or bug somewhere that occasionally would result in borking the test run to the point where local retries don't even help.

**Pull request merged**